Lib/Protocols/IdPOP3.pas

```diff
@@ -619,6 +631,7 @@ procedure TIdPOP3.Connect;
     if FAutoLogin then begin
       Login;
     end;
+    SendUTF8IfAdvertised;

```

Comment:
Sending the `UTF8` command at this location is a Catch-22. `AutoLogin` is True by default, so you would typically be sending the `UTF8` command after `Login()` has already sent the `STLS` and `USER`/`PASS`/`APOP`/`AUTH` commands, which is too late for sending UTF-8 encoded authentication.  On the other hand, the `UTF8` command is not allowed to be sent before an `STLS` command, so you can't just move the call before `Login()`.

So, you are going to have to move the `UTF8` command inside of `Login()` itself instead, so that `Login()` can send the `UTF8` command after sending the `STLS` command and before sending the `USER`/`PASS`/`APOP`/`AUTH` commands.  If `AutoLogin` is False, `Connect()` should not send `UTF8` at all, as that would then prevent a later call to `Login()` from sending the `STLS` command.

The alternative would be to manually keep track of whether `TIdPOP3` is currently operating in ASCII or UTF8 mode and then send the `UTF8` command only when absolutely necessary for a subsequent command to use UTF-8.